apps/buildog/src/components/org-navigation.tsx

```diff
@@ -98,8 +99,11 @@ export default function OrgNavigation() {
                     key={org.organization_id}
                     value={org.organization_id.toString()}
                     onSelect={() => {
-                      router.push(`/organizations/${org.organization_id}`);
-                      setOpen(false);
+                      const newPath = pathname.replace(
+                        organizationId as string,
+                        org.organization_id.toString()
+                      );
+                      router.replace(newPath);
```

Comment:
1. Can we move this to another function. `handleOrganizationRouteChange`
2. Could you please change .replace to .push [resource link](https://nextjs.org/docs/14/app/api-reference/functions/use-router)
3. Remove the "as string" and start with type check (ex.  `if (typeof organizationId === 'string')`)
4. remove the .toString on org.organizartion_id. org.organization_id should be string already.

```
  const handleOrganizationRouteChange = (newOrganizationId: string) => {
        if (typeof organizationId === 'string') {
            const newPath = pathname.replace(
                organizationId,
                newOrganizationId
            );
            router.replace(newPath);
        }
    }
```